### DISCUSSION TITLE
Bill not connecting to expenses transaction

### DISCUSSION THREAD

**Discussion by thehijacker** [firefly-iii/firefly-iii#8156] [Firefly III]
Hello,

I have few bills defined that could be auto created or I will input them manually every month. For all that I have already created and there is an expense transaction, I can also see the next expanse under "Paid this period". All but one. I wish to figure out why:

Bill:

![image](https://github.com/firefly-iii/firefly-iii/assets/13857043/b0eb8b4f-19df-42e2-a5ca-55cde498ebc0)

Rule:

![image](https://github.com/firefly-iii/firefly-iii/assets/13857043/c58a2ea9-6bd9-4978-9959-8a9603dd7eb9)

And expanse:

![image](https://github.com/firefly-iii/firefly-iii/assets/13857043/e52f62f5-b0e4-4624-8955-164bf9a3b143)

Under rule if I click "See matching transactions" it will find it. So why isn't it listed under Paid this period under Bills?

**Comment by JC5** [comment]
Have you tried executing the rule?

**Comment by thehijacker** [comment]
Yes I have. It finds nothing to update:

![image](https://github.com/firefly-iii/firefly-iii/assets/13857043/9787fd69-8394-4599-a9e5-22944baa28ab)

But If I select "Select matching transactions" it finds one.

**Reply by thehijacker** [reply]
The transaction is dated 16.11. but rule only wishes to apply transactions up to 15.11.2023. Not sure why:

![image](https://github.com/firefly-iii/firefly-iii/assets/13857043/efd6623e-d27f-43ab-9274-09ee660516a0)

**Reply by thehijacker** [reply]
So is this a bug or I have configuration error somewhere?

**Reply by JC5** [reply]
I see the rule form does not accept future dates. Other than that, I can't say. When I test your rule it works fine.

**Reply by thehijacker** [reply]
I will see how it goes next month when I input the bills. If again it will not match only one in future.
